.github/workflows/clp-core-build.yaml

```diff
@@ -61,32 +43,71 @@ jobs:
         uses: "dorny/paths-filter@v2"
         id: "filter"
         with:
-          base: "${{env.PATHS_FILTER_BASE}}"
+          # Consider changes between the current commit and `main`
+          # NOTE: If a pull request changes the image, then we need to build the image and then
+          # build CLP using the changed image; otherwise we can build CLP using the published image.
+          # So when determining what files have changed, we use `main` rather than the previous
+          # commit to ensure that on every push to the PR, we can detect if the image was changed
+          # and use it to build and test CLP.
```

Comment:
Maybe make it clearer that the published image is based off main and the commits are just stored temporarily with github artifact